common/src/main/java/org/dash/wallet/common/util/security/EncryptionProvider.kt

```diff
@@ -1,83 +1,33 @@
+/*
+ * Copyright 2023 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 package org.dash.wallet.common.util.security
 
-import android.content.SharedPreferences
-import android.security.keystore.KeyGenParameterSpec
-import android.security.keystore.KeyProperties
-import android.util.Base64
-import java.nio.charset.StandardCharsets
+import java.io.IOException
 import java.security.GeneralSecurityException
-import java.security.KeyStore
 import java.security.KeyStoreException
-import javax.crypto.Cipher
-import javax.crypto.KeyGenerator
-import javax.crypto.SecretKey
-import javax.crypto.spec.GCMParameterSpec
-
-class EncryptionProvider(
-    private val keyStore: KeyStore,
-    private val securityPrefs: SharedPreferences
-) {
-    private val ENCRYPTION_IV_KEY = "encryption_iv"
-    private var encryptionIv = restoreIv()
-
-    private val cipher by lazy {
-        Cipher.getInstance("AES/GCM/NoPadding")
-    }
 
-    @Throws(GeneralSecurityException::class)
-    fun encrypt(keyAlias: String, textToEncrypt: String): ByteArray? {
-        val secretKey = getSecretKey(keyAlias)
-        if (encryptionIv == null) {
-            cipher.init(Cipher.ENCRYPT_MODE, secretKey)
-            saveIv(cipher.iv)
-        } else {
-            val spec = GCMParameterSpec(128, encryptionIv)
-            cipher.init(Cipher.ENCRYPT_MODE, secretKey, spec)
-        }
-        return cipher.doFinal(textToEncrypt.toByteArray(StandardCharsets.UTF_8))
-    }
+interface EncryptionProvider {
```

Comment:
Keeping the interface only in `common` and providing the implementation in the `wallet`.